From ddec088efc61dcc0f507013c08500b6319f68e2f Mon Sep 17 00:00:00 2001
From: Daniel Mach <daniel.mach@suse.com>
Date: Mon, 22 Aug 2022 15:45:25 +0200
Subject: [PATCH] Replace imp with importlib

---
 osc/commandline.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/osc/commandline.py b/osc/commandline.py
index 12ca969a3..686a2d4de 100644
--- a/osc/commandline.py
+++ b/osc/commandline.py
@@ -3,7 +3,7 @@
 import sys
 import signal
 import time
-import imp
+import importlib.util
 import inspect
 import os
 try:
@@ -9431,8 +9431,11 @@ def _load_plugins(self):
                 if not extfile.endswith('.py'):
                     continue
                 try:
-                    modname = os.path.splitext(extfile)[0]
-                    mod = imp.load_source(modname, os.path.join(plugin_dir, extfile))
+                    modname = "osc.plugins." + os.path.splitext(extfile)[0]
+                    spec = importlib.util.spec_from_file_location(modname, os.path.join(plugin_dir, extfile))
+                    mod = importlib.util.module_from_spec(spec)
+                    sys.modules[modname] = mod
+                    spec.loader.exec_module(mod)
                     # restore the old exec semantic
                     mod.__dict__.update(globals())
                     for name in dir(mod):
